account/recovery.go

```diff
@@ -188,17 +192,254 @@ func recoverInitalState(ctx context.Context, cfg RecoveryConfig) (
 	return accounts, nil
 }
 
+// recreatePossibleAccounts returns a set of potentially valid accounts
+// by generating a set of recovery keys.
+func recreatePossibleAccounts(ctx context.Context, cfg RecoveryConfig) (
+	map[*Account]struct{}, error) {
+
+	possibleAccounts := make(map[*Account]struct{}, cfg.AccountTarget)
+
+	// Prepare the keys we are going to try. Possibly not all of them will
+	// be used.
+	KeyDescriptors, err := GenerateRecoveryKeys(
+		ctx, cfg.AccountTarget, cfg.Wallet,
+	)
+	if err != nil {
+		return possibleAccounts, fmt.Errorf(
+			"error generating keys: %v", err,
+		)
+	}
+
+	for _, keyDes := range KeyDescriptors {
+		secret, err := cfg.Signer.DeriveSharedKey(
+			ctx, cfg.auctioneerPubKey, &keyDes.KeyLocator,
+		)
+		if err != nil {
+			return possibleAccounts, fmt.Errorf(
+				"error deriving shared key: %v", err,
+			)
+		}
+
+		acc := &Account{
+			TraderKey:     keyDes,
+			AuctioneerKey: cfg.auctioneerPubKey,
+			Secret:        secret,
+		}
+		possibleAccounts[acc] = struct{}{}
+	}
+
+	return possibleAccounts, nil
+}
+
+// findAccounts tries to find on-chain footprint for the creation of a set
+// of possible pool accounts.
+func findAccounts(cfg RecoveryConfig,
```

Comment:
I think we should add some test cases for this.

Reply:
I added a couple of tests for this, I will add another one for the "findAccountUpdate"